### PR TITLE
Clarify optional {BIB}, warn about curly quotes

### PR DESCRIPTION
**Pull request by markcmiller86**
* [x] Set list of Reviewers (at least one).
* [x] Add to Project [BSSw Internal].
* [x] View the modified `*.md` files as rendered in GitHub.
* [ ] If changes are to the GitHub pages site under the `docs/` directory, consider viewing locally with Jekyll.
* [x] Watch for PR check failures.
* [x] Make any final changes to the PR based on feedback and review GitHub (and Jekyll) rendered files.
* [x] Ensure at least one reviewer signs off on the changes.
* [ ] Once reviewer has approved and PR check pass, then merge the PR.

[preview]: https://preview.bssw.io
[bssw.io]: https://bssw.io
[Content Development]: https://github.com/betterscientificsoftware/bssw.io/projects/3
[BSSw Internal]: https://github.com/betterscientificsoftware/bssw.io/projects/2
[meta-data]: https://betterscientificsoftware.github.io/bssw.io/bssw_styling_common.html#metadata-section
[wikize_refs.py]: https://github.com/betterscientificsoftware/bssw.io/blob/master/utils/README.md#wikize_refspy

### CODE REVIEW

File: Articles/Blog/ReferencesInMarkdownHybridApproach.md [hunk 1]
```diff
@@ -39,10 +39,9 @@ at the end of the markdown file, of the form
   - **Note**: The post-processing script renumbers them 1...N
 - `URL`: (**REQUIRED**) The URL for the reference
 - `TIT`: (*optional*) A title for the reference (appears as tool-tip during hover)
-- `BIB`: (*optional*) Full bibliographic information for the reference
-  - Here, we are using the `{BIB}` field, embedded between `{` and `}` in the
-    [link's title](https://www.markdownguide.org/basic-syntax#adding-titles) text
-    as a notational *extension*.
+- `{BIB}`: (*optional*) Full bibliographic information for the reference
+  - Here, we are using the `{BIB}` field, embedded in the [link's title](https://www.markdownguide.org/basic-syntax#adding-titles) text as a notational *extension*.
+- **Note**: Be sure whatever editor you are using is not inadvertently sneaking in [*smart quotes* or *curly quotes*](https://practicaltypography.com/straight-and-curly-quotes.html) and that you are using only *straight* quotes.
```
Comment (bartlettroscoe) [inline Articles/Blog/ReferencesInMarkdownHybridApproach.md:44]: Yea, you have to be very careful when doing copy and paste from some Word document on web browser into an editor that copies these Unicode chars over.

Actually, a Markdown linter not look for these non-*straight* quotes and warn you about them?  I can't seem to find a rule like this at:

* https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md

Seems like we might need to write our own checker for that.  (Would not be hard to do that in a simple Python script.)

**Review by bartlettroscoe — APPROVED** [review]
Good warning.  It would be great if we could catch this automatically with a linter (see comment below).

File: Articles/Blog/ReferencesInMarkdownHybridApproach.md (hunk 1 above)
Comment (markcmiller86) [reply Articles/Blog/ReferencesInMarkdownHybridApproach.md:44]: > you have to be very careful when doing copy and paste

Good point. I added an additional note about that

**Review by bartlettroscoe — APPROVED** [review]
LGTM

**Pull request merged**